### PR TITLE
Add a helper command for one_gadget

### PR DESCRIPTION
**Pull request by lebr0nli**
This PR adds a new command for finding valid [one_gadget](https://github.com/david942j/one_gadget) for the current context and makes it easier to debug when the constraints are unsat.

----

Here are some example usages:

After we control pc, if we get some luck, we might be able to find a satisfiable gadget via command: `onegadget`:
image

If there's no satisfiable one_gadget for the current context, we can run with `--show-unsat` to find out what constraint is unsatisfiable: 
image

If we want to know more information behind the unsatisfiable gadgets, we can run with `--verbose` or `-v`:
image

### CODE REVIEW

**Comment by disconnect3d** [comment]
Any chance we can display the condition and SAT/UNSAT in two columns? Or somehow that it will be more readable? I feel like the output is bloated with colors and maybe not so readable now.

We could also rename the command to just `onegadget` too 🤔

**Comment by lebr0nli** [comment]
> Any chance we can display the condition and SAT/UNSAT in two columns? Or somehow that it will be more readable? I feel like the output is bloated with colors and maybe not so readable now.

Yeah, I also feel that the output is a little bit messy at the moment.
I'll attempt to organize it into a table format with columns to see if that offers a cleaner presentation.

> We could also rename the command to just `onegadget` too 🤔

Sure!

**Comment by lebr0nli** [comment]
Regarding the output format, how about something like this?
SCR-20240219-mdcn

^ I think this looks much better(?), but it sacrifices the ability to display the detailed information behind its UNSAT status.
This results that I still need to check each UNSAT constraint manually to find out how to make it SAT when debugging.

Should we use the tabulate output as default, but add an `-v` option to show the original messy but detailed output?
Or any other ideas?

**Comment by disconnect3d** [comment]
> ^ I think this looks much better(?), but it sacrifices the ability to display the detailed information behind its UNSAT status. This results that I still need to check each UNSAT constraint manually to find out how to make it SAT when debugging.
> 
> Should we use the tabulate output as default, but add an `-v` option to show the original messy but detailed output? Or any other ideas?

This looks good

**Comment by lebr0nli** [comment]
> This looks good

Ok!
I'll update the output format to the tabulate one.

File: pwndbg/commands/one_gadget.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import pwndbg.glibc
+from pwndbg.commands import CommandCategory
+
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.RawTextHelpFormatter,
+    description="""Show one_gadget
+
+Examples:
+    one_gadget
+    one_gadget --show-unsat
```
Comment (anthraxx) [inline pwndbg/commands/one_gadget.py]: The command has been renamed to `onegadget`

File: pwndbg/commands/one_gadget.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import pwndbg.color.message as M
+import pwndbg.commands
+import pwndbg.gdblib.one_gadget
+import pwndbg.glibc
+from pwndbg.commands import CommandCategory
+
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.RawTextHelpFormatter,
+    description="""Show one_gadget
```
Comment (anthraxx) [inline pwndbg/commands/one_gadget.py]: The command has been renamed to `onegadget`

Comment (disconnect3d) [reply pwndbg/commands/one_gadget.py]: @lebr0nli perhaps also rename the file to `onegadget.py`

File: pwndbg/commands/one_gadget.py (hunk 1 above)
Comment (lebr0nli) [reply pwndbg/commands/one_gadget.py]: Ahh, yes, thanks, I'll fix it!

File: pwndbg/commands/one_gadget.py (hunk 2 above)
Comment (lebr0nli) [reply pwndbg/commands/one_gadget.py]: Sure, I'll rename it

File: pwndbg/gdblib/onegadget.py [hunk 3]
```diff
@@ -0,0 +1,595 @@
...
+
+ADDRESS_WRITABLE_PATTERN = re.compile(r"address(?:es)? (.*) (?:is|are) writable")
+WRITABLE_COLON_PATTERN = re.compile(r"writable: (.*)")
+EQUAL_NULL_PATTERN = re.compile(r"(.*) == NULL")
+VALID_POSIX_SPAWN_FILE_ACTIONS_PATTERN = re.compile(r"(.*) <= 0")
+VALID_ARGV_PATTERN = re.compile(r"(.+) is a valid argv")
+VALID_ENVP_PATTERN = re.compile(r"(.+) is a valid envp")
+IS_ALIGNED_PATTERN = re.compile(r"(.+) & 0xf == (\d+)")
+IS_GOT_ADDRESS_PATTERN = re.compile(r"(.+) is the GOT address of libc")
+CAST_PATTERN = re.compile(r"^\([s|u]\d+\)")
```
Comment (disconnect3d) [inline pwndbg/gdblib/onegadget.py:32]: Fwiw all this `re.compile` etc. will slow down Pwndbg load time, but it seems negligible:
```
pwndbg> !cat a.py
import re
import time

a = time.time()
ADDRESS_WRITABLE_PATTERN = re.compile(r"address(?:es)? (.*) (?:is|are) writable")
WRITABLE_COLON_PATTERN = re.compile(r"writable: (.*)")
EQUAL_NULL_PATTERN = re.compile(r"(.*) == NULL")
VALID_POSIX_SPAWN_FILE_ACTIONS_PATTERN = re.compile(r"(.*) <= 0")
VALID_ARGV_PATTERN = re.compile(r"(.+) is a valid argv")
VALID_ENVP_PATTERN = re.compile(r"(.+) is a valid envp")
IS_ALIGNED_PATTERN = re.compile(r"(.+) & 0xf == (\d+)")
IS_GOT_ADDRESS_PATTERN = re.compile(r"(.+) is the GOT address of libc")
CAST_PATTERN = re.compile(r"^\([s|u]\d+\)")
b = time.time()

elapsed = b - a
print(elapsed)
pwndbg> source a.py
0.0009448528289794922
```

**Pull request merged**